src/main/java/org/alliancegenome/curation_api/services/DataProviderService.java

```diff
@@ -44,19 +57,100 @@ public DataProvider createAffiliatedModDataProvider() {
 	public DataProvider getAllianceDataProvider() {
 		return getDefaultDataProvider("Alliance");
 	}
-	
+
 	@Transactional
 	public DataProvider getDefaultDataProvider(String sourceOrganizationAbbreviation) {
 		return dataProviderDAO.getOrCreateDataProvider(organizationDAO.getOrCreateOrganization(sourceOrganizationAbbreviation));
 	}
 
+	@Transactional
+	public ObjectResponse<DataProvider> insertExpressionAtlasDataProvider(DataProvider entity) {
+		String referencedCurie = entity.getCrossReference().getReferencedCurie();
+		// find associated gene
+		Long geneID = getAssociatedGeneId(referencedCurie, entity.getSourceOrganization());
+		// if no gene found skip (= don't import) the accession
+		if (geneID == null) {
+			return new ObjectResponse<>();
+		}
+
+		DataProvider dbEntity = getDataProvider(entity.getSourceOrganization(), referencedCurie, entity.getCrossReference().getResourceDescriptorPage());
+		// we only create new records, no updates
+		if (dbEntity == null) {
+			dataProviderDAO.persist(entity);
+			if (!List.of("FB", "SGD").contains(entity.getSourceOrganization().getAbbreviation())) {
+				crossReferenceDAO.persistAccessionGeneAssociated(entity.getCrossReference().getId(), geneID);
+			}
+			return new ObjectResponse<>(entity);
+		}
+		return new ObjectResponse<>(dbEntity);
+	}
+
+	@NotNull
+	public static String getFullReferencedCurie(String localReferencedCurie) {
+		return RESOURCE_DESCRIPTOR_PREFIX + ":" + localReferencedCurie;
+	}
+
+	Map<String, Long> accessionGeneMap = new HashMap<>();
+	public static final String RESOURCE_DESCRIPTOR_PREFIX = "ENSEMBL";
+	public static final String RESOURCE_DESCRIPTOR_PAGE_NAME = "default";
```

Comment:
Can these variable be moved to the top of the file so that one does not have to go hunting for them.